index.bs

```diff
@@ -232,27 +232,130 @@ Non-Functional Requirements {#requirements-non-functional}
+  metadata.  The time value is represented as zero or a positive integer.  The
+  clock and timescale are chosen by the advertising endpoint, but are not
+  important, except that the clock must be monotonic.  A larger value is an
+  indication to discovering clients that they should reconnect to the
+  advertising client to learn updated metadata.  
+ 
+If an endpoint wishes to advertise support for more than one version of the
+protocol, it may do so by advertising more than one DNS-SD entry, which may have
+the same DNS TXT fingerprint value.
+
 
-Transport Establishment {#transport}
+Transport and metadata discovery with QUIC {#transport}
 ====================================
 
-Issue(101): Incorporate material from the \[QUIC](quic.md) document.
+If a discovering endpoint wants to connect to an advertising endpoint, or to
+learn further metdata about it, it initiates a
+[QUIC](https://tools.ietf.org/html/draft-ietf-quic-transport-16) connection to
+the IP and port of the mDNS entry.  Prior to authentication, some message may be
+exchanged (such as further metadata), but such info should be treated as
+unauthenticated (such as indicating to a user that a display name of an
+unauthenticated endpoint is unauthenticated).  
+
+To learn further metadata, an endpoint may send a endpoint-info-request
+message (see Appendix A) and receive back a endpoint-info-response message.  The
+messages may contain the following information with the following meaning:
+
+- display-name: The display name of the responding endpoint intended to be
+  displayed to a user by the requesting endpoint.  If the responding endpoint is
+  not yet authenticated, the requesting endpoint should make UI affordance for
+  indicating to the user that the display name is not yet authenticated.  If the
+  responding endpoint changes its display name, the requesting endpoint should
+  make UI affordance for indicating to the user that the display name has
+  changed. 
+
+- model-name: The model name of the device (if it is a device), used mainly for
+  debugging purposes, but may be displayed to the user of the requesting
+  endpoint.
+
+<!-- TODO: Add device type and/or capabilities -->
+
+If a discovering endpoint (acting as a QUIC client) wishes to receive messages
+from an advertising endpoint (acting as a QUIC server) or an advertising
+endpoint (acting as a QUIC server) wishes to send message to a discovering
+endpoint (acting as a client endpoint), it may wish to keep the QUIC connection
+alive.  Once neither side needs to keep the connection alive for the purposes of
+sending or receiving messages, the connection should be closed.  In order to
+keep a QUIC connection alive, an endpoint may send an endpoint-status-request
+message, and any endpoint that receives an endpoint-status-request message
+should send an endpoint-status-response message.  Such messages should be sent
+every 25 seconds.  If, however, other mechanisms or traffic (such as other
```

Comment:
I just saw that under the `PING` frame description, the answer is basically "yes, but..." It sounds like `idle_timeout` would only be useful if it would lower our 25 second timer.

Reply:
That's a good point about the idle_timeout transport parameter.  Perhaps we should just say "send more frequently than idle_timeout" and not specify 25s (since the QUIC spec recommends 15-30 seconds).